opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Wise.kt

```diff
@@ -0,0 +1,563 @@
+        val titleId = id!!.split(":")[0]
+        return "$baseurl/wise-apps/catalog/$homeBranch/detail/wise/$titleId"
+    }
+
+    override fun filterResults(filter: Filter?, option: Filter.Option?): SearchRequestResult {
+        TODO("Not yet implemented")
+    }
+
+    override fun getSupportedLanguages(): MutableSet<String> {
+        val langs: MutableSet<String> = HashSet()
+        if (libraryData.has("languages")) {
+            try {
+                for (i in 0 until libraryData.getJSONArray("languages").length()) {
+                    langs.add(libraryData.getJSONArray("languages").getString(i))
+                }
+            } catch (e: JSONException) {
+                e.printStackTrace()
+            }
+        }
+        return langs
+    }
+
+    override fun parseSearchFields(): List<SearchField> {
+        if (!initialised) start()
+
+        val freeSearchField = BarcodeSearchField("search", stringProvider.getString("nav_search"), false,false,"")
+        val ietsTitelOrAuteur = DropdownSearchField().apply {
+            id = "searchScope"
+            displayName = stringProvider.getString("search_scope")
+            dropdownValues = listOf(
+                    DropdownSearchField.Option("iets", stringProvider.getString("all")),
+                    DropdownSearchField.Option("titel", stringProvider.getString("title")),
+                    DropdownSearchField.Option("auteur", stringProvider.getString("author")))
+            isAdvanced = false
+        }
+
+
+        val materiaal = DropdownSearchField().apply {
+            id = "materiaal"
+            displayName = stringProvider.getString("mediatype")
+            dropdownValues = listOf(
+                    DropdownSearchField.Option("", stringProvider.getString("all")),
+                    DropdownSearchField.Option("BOE", stringProvider.getString("mediatype_book")),
+                    DropdownSearchField.Option("GBK", stringProvider.getString("mediatype_audiobook")),
+                    DropdownSearchField.Option("CD", stringProvider.getString("mediatype_cd")),
+                    DropdownSearchField.Option("DVD", stringProvider.getString("mediatype_dvd"))
+            )
+            isAdvanced = false
+        }
+        if ( searchLocations.size > 1 ) {
+            val locations = DropdownSearchField().apply {
+                id = "location"
+                displayName = stringProvider.getString("location")
+                dropdownValues = searchLocations.map { (first, last) -> DropdownSearchField.Option(first, last)  }
+                isAdvanced = false
+            }
+            return listOf(freeSearchField, ietsTitelOrAuteur, locations, materiaal)
+        }
+        return listOf(freeSearchField, ietsTitelOrAuteur, materiaal)
+
+    }
+
+
+    override fun getSupportFlags(): Int {
+        return OpacApi.SUPPORT_FLAG_ENDLESS_SCROLLING or OpacApi
+                .SUPPORT_FLAG_WARN_RESERVATION_FEES or OpacApi.SUPPORT_FLAG_ACCOUNT_PROLONG_ALL
```

Comment:
You should not set SUPPORT_FLAT_ACCOUNT_PROLONG_ALL if you don't actually implement it ;)

Reply:
Oops!